src/interaction/LabelsManager.cpp

```diff
@@ -98,15 +99,15 @@ void LabelsManager::add(std::string &texture_path, float x, float y, float z, fl
   if (m_texture_cache.find(texture_path) == m_texture_cache.end())
   {
 
-    Texture *texture = new Texture(GL_TEXTURE_2D, texture_path);
+      std::shared_ptr<Texture> texture = std::make_shared< Texture>(GL_TEXTURE_2D, texture_path);
```

Comment:
And this means their can be multiple copies of the same pointer in the app?

Reply:
Yup. All is managed by the compiler.